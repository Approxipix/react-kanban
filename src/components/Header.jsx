import React, { Component } from 'react'
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { NavLink } from 'react-router-dom';
import logo from '../images/logo.svg'
import { connect } from "react-redux";

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 1.5rem;
  background-color: ${props => props.color};
  box-shadow: inset 0px 0px 0px 40px rgba(0, 0, 0, .3)
`;

const Nav = styled.nav`

`;

const NavList = styled.ul`
  display: flex;
`;

const NavItem = styled.li`
  display: flex;
  align-items: center;
  margin-right: .5rem
  padding: .3rem .5rem
  color: rgba(255, 255, 255, 0.8);
  border-radius: 3px;
  background-color: rgba(255, 255, 255, 0.2);
  transition: color .2s ease-in-out;
  &:hover {
    color: #fff;
   }
  }
`;

const NavIcon = styled.i`
  margin-right: .3rem;
  font-size: 1.2rem;
`;

const NavTitle = styled.div`
  font-size: 1rem;
  line-height: 1.5;
`;

const Logo = styled.img`
  width: 10rem;
`;

const Actions = styled.div`
`;

class Header extends Component {
  render() {
    const { boards } = this.props;
    let boardIndex;
    const board = boards.find((board, index) => {
      boardIndex = index;
      return board.boardId === 0;
    });
    return (
      <Wrapper color={board.color}>
        <Nav>
          <NavList>
            <NavLink to="/boards">
              <NavItem>
                <NavIcon>
                  <FontAwesomeIcon icon="columns" />
                </NavIcon>
                <NavTitle>
                  Boards
                </NavTitle>
              </NavItem>
            </NavLink>
          </NavList>
        </Nav>
        <Logo src={logo} />
        <Actions />
      </Wrapper>
    )
  }
}

function mapStateToProps(state) {
  return {
    boards: state.rootReducer.boards,
  }
}

export default connect(mapStateToProps)(Header);
