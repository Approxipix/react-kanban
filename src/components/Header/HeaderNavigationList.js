const HeaderNavigationList = [
  {
    title: 'Boards',
    path: '/',
    faIcon: 'columns',
  },
];

export default HeaderNavigationList
