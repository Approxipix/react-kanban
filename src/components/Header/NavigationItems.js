const NavigationItems = [
  {
    title: 'Boards',
    path: '/boards',
    faIcon: 'columns',
  },
];

export default NavigationItems
