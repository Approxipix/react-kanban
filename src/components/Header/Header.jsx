import React, { Component } from 'react'
import { connect } from 'react-redux';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome/index';
import NavigationItems from './NavigationItems';
import logo from '../../images/logo.svg'

const Wrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 1rem;
  align-items: center;
  padding: 0 1.5rem;
  background-color: ${props => props.color};
  box-shadow: inset 0px 0px 0px 40px rgba(0, 0, 0, .3);
  transition: background-color .2s ease-in-out;
`;

const Nav = styled.nav``;

const NavList = styled.ul`
  display: flex;
`;

const NavItem = styled.li`
  display: flex;
  align-items: center;
  margin-right: .5rem
  padding: .3rem .5rem
  color: rgba(255, 255, 255, 0.8);
  border-radius: 3px;
  background-color: rgba(255, 255, 255, 0.2);
  transition: all .2s ease-in-out;
  &:hover {
    color: #fff;
    background-color: rgba(255, 255, 255, 0.25);
  }
`;

const NavIcon = styled.i`
  margin-right: .3rem;
  font-size: 1.2rem;
`;

const NavTitle = styled.div`
  font-size: 1rem;
  line-height: 1.5;
`;

const Logo = styled.img`
  width: 10rem;
  margin: 0 auto;
`;

const Actions = styled.div``;

class Header extends Component {
  render() {
    const { boardColor } = this.props;
    return (
      <Wrapper color={boardColor}>
        <Nav>
          {NavigationItems.map((item, index) => {
            if (window.location.pathname === item.path) return null;
            return (
              <NavList key={index}>
                <NavLink to={item.path}>
                  <NavItem>
                    <NavIcon>
                      <FontAwesomeIcon icon={item.faIcon} />
                    </NavIcon>
                    <NavTitle>
                      {item.title}
                    </NavTitle>
                  </NavItem>
                </NavLink>
              </NavList>
            )
          })}
        </Nav>
        <Logo src={logo} />
        <Actions />
      </Wrapper>
    )
  }
}

function mapStateToProps(state) {
  const board = state.boardReducer[state.rootReducer.currentBoardID];
  return {
    boardColor: !!board ? board.color : state.rootReducer.colors[0]
  }
}

export default connect(mapStateToProps)(Header);
