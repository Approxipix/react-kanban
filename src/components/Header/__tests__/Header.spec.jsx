import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import Header from '../Header';

const mockStore = configureStore([]);

describe('<Header>', () => {
  let store;
  let wrapper;
  let component;

  beforeEach(() => {
    store = mockStore({
      rootReducer: {
        currentBoardID: "Board2ID"
      },
      boardReducer: {
        Board1ID: { _boardId: "Board1ID", color: "61BC4F" },
        Board2ID: { _boardId: "Board2ID", color: "FF9F1A" }
      }
    });
    const state = {
      navigation: [
        { title: 'Home', path: '/home', faIcon: 'times', },
        { title: 'Boards', path: '/boards', faIcon: 'columns', },
      ]
    };

    wrapper = shallow(<Header store={store}  />);
    component = wrapper.find('Header').dive();
    component.setState({ ...state });
  });

  it('should render with given state from Redux store', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should sets the color prop as the `value` if the _boardID and currentBoardID are the same', () => {
    expect(component.props().color).toEqual("FF9F1A");
  });

  it('should render a nav link for each item from state', () => {
    expect(component.find('NavList').length).toEqual(2)
  });
});
