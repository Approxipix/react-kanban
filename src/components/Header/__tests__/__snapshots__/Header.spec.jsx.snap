// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render with given state from Redux store 1`] = `
<styled.div
  color="FF9F1A"
>
  <styled.nav>
    <NavList
      key="0"
    >
      <NavLink
        to="/home"
      >
        <styled.li>
          <styled.i>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="times"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </styled.i>
          <styled.h3>
            Home
          </styled.h3>
        </styled.li>
      </NavLink>
    </NavList>
    <NavList
      key="1"
    >
      <NavLink
        to="/boards"
      >
        <styled.li>
          <styled.i>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="columns"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
          </styled.i>
          <styled.h3>
            Boards
          </styled.h3>
        </styled.li>
      </NavLink>
    </NavList>
  </styled.nav>
  <styled.img
    src="logo.svg"
  />
  <styled.div />
</styled.div>
`;
